coding-standards/php.md

```diff
@@ -1,15 +1,322 @@
+	<rule ref="SlevomatCodingStandard.TypeHints.UselessConstantTypeHint" />
+	<rule ref="SlevomatCodingStandard.Arrays.DisallowImplicitArrayCreation" />
+	<rule ref="SlevomatCodingStandard.ControlStructures.RequireNullCoalesceOperator" />
+	<rule ref="SlevomatCodingStandard.PHP.UselessSemicolon" />
+	<rule ref="SlevomatCodingStandard.Variables.UnusedVariable">
+		<properties>
+			<property name="ignoreUnusedValuesWhenOnlyKeysAreUsedInForeach" value="true" />
+		</properties>
+	</rule>
+	<rule ref="SlevomatCodingStandard.Variables.UselessVariable" />
+	<rule ref="SlevomatCodingStandard.Exceptions.DeadCatch" />
+	<rule ref="SlevomatCodingStandard.Classes.MethodSpacing" >
+		<properties>
+			<property name="minLinesCount" value="1"/>
+			<property name="maxLinesCount" value="1"/>
+		</properties>
+	</rule>
+	<rule ref="SlevomatCodingStandard.Classes.PropertySpacing">
+		<properties>
+			<property name="minLinesCountBeforeWithComment" value="0"/>
+			<property name="maxLinesCountBeforeWithComment" value="0"/>
+			<property name="minLinesCountBeforeWithoutComment" value="0"/>
+			<property name="maxLinesCountBeforeWithoutComment" value="0"/>
+		</properties>
+	</rule>
+	<rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
+		<properties>
+			<property name="searchAnnotations" value="true"/>
+		</properties>
+	</rule>
+	<rule ref="SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses" />
+	<rule ref="SlevomatCodingStandard.Whitespaces.DuplicateSpaces" />
+	<rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing">
+		<properties>
+			<property name="spacesCountBeforeColon" value="0"/>
+		</properties>
+	</rule>
+	<rule ref="SlevomatCodingStandard.Commenting.DisallowCommentAfterCode" />
+	<rule ref="SlevomatCodingStandard.Commenting.EmptyComment" />
+	<rule ref="SlevomatCodingStandard.Commenting.RequireOneLineDocComment" />
+	<rule ref="SlevomatCodingStandard.ControlStructures.UselessIfConditionWithReturn" />
+	<rule ref="SlevomatCodingStandard.ControlStructures.UselessTernaryOperator" />
+
+    <!-- disable useless function comment -->
+    <rule ref="SlevomatCodingStandard.Commenting.UselessFunctionDocComment" />
+</ruleset>
+```
+#### Usage
+
+Add couple of script commands to run `phpcs` and `phpcbf` in `composer.json` file.
+
+```json
+"scripts": {
+    // ..<existing scripts>
+    "lint": "vendor/bin/phpcs --standard=.phpcs.xml",
+    "lint:fix": "vendor/bin/phpcbf --standard=.phpcs.xml",
+  },
+```
+
+**`composer run lint`**
+
+This command will tokenize PHP, JavaScript and CSS files to detect violations of a defined coding standard.
+
+**`composer run lint:fix`**
+
+This command will automatically correct coding standard violations.
+
+### PHP PHAN
+#### Installation 
+
+```json
+composer require phan/phan --dev
+```
+Inorder to let your project use php phan, you have to create a folder `.phan` and a file in it `config.php`. All the settings for phan are placed there. 
```

Comment:
In some places we are using PHP phan, and in others php phan...lets try to be uniform.